### DISCUSSION TITLE
PorousFlow sinks

### DISCUSSION THREAD

**Discussion by MatiasAllay** [idaholab/moose#18259] [Q&A Modules: General] [answered]
I'm sorry for bringing another simple question...
I'm modeling Theis problem but instead of `PorousFlowSquarePulsePointSource` I would like to use `PorousFlowSink`. However, for some reason nothing happens:

[Mesh]
  type = GeneratedMesh
  dim = 2
  nx = 20
  ny = 1
  ymax = 1
  xmax = 100
  bias_x = 1.05
[]

[Problem]
  coord_type = RZ
[]

[GlobalParams]
  PorousFlowDictator = dictator
[]

[Variables]
  [pp]
    initial_condition = 20E6
  []
[]

[PorousFlowBasicTHM]
  dictator_name = dictator
  add_darcy_aux = false
  fp = simple_fluid
  gravity = '0 0 0'
  multiply_by_density = false
  porepressure = pp
[]

[Modules]
  [FluidProperties]
    [simple_fluid]
      type = SimpleFluidProperties
      viscosity = 0.001
    []
  []
[]

[Materials]
  [porosity]
    type = PorousFlowPorosityConst
    porosity = 0.05
  []
  [biot_mod]
    type = PorousFlowConstantBiotModulus
    fluid_bulk_modulus = 2E9
    biot_coefficient = 1.0
  []
  [permeability]
    type = PorousFlowPermeabilityConst
    permeability = '1E-14 0 0 0 1E-14 0 0 0 1E-14'
  []
[]

[BCs]
  [sink]
    type = PorousFlowSink
    variable = pp
    flux_function = 0.16E-3 
    boundary = left
  []
[]

[VectorPostprocessors]
  [pp]
    type = LineValueSampler
    num_points = 25
    start_point = '0 0 0'
    end_point = '100 0 0'
    sort_by = x
    variable = pp
  []
[]

[Preconditioning]
  [smp]
    type = SMP
    full = true
  []
[]

[Executioner]
  type = Transient
  solve_type = Newton
  dt = 200
  end_time = 1E3
  nl_abs_tol = 1e-10
[]

[Outputs]
 
  [csv]
    type = CSV
    execute_on = final
  []
[]

Also, how is the area of the sink defined in this case? Would it be 2 * pi * Radius * height?

Thanks MOOSE
Matias

**Comment by cpgr** [comment, marked as answer]
Hi Matias,

It is most likely due to the zero surface area for the sink at `r = 0`. Shift the mesh slightly (eg `xmin = 0.1` in the `Mesh` block) and it should work (injection is through the area of the cylinder like you had above). You can use an `AreaPostProcessor` to compute the surface area of the sink for you and compute the flux for your sink in a function.

**Reply by cpgr** [reply]
There is an example of this in https://github.com/idaholab/moose/blob/06bc1f73f0d4dfb3d01ce9ba77d6bb97346e07eb/modules/porous_flow/examples/restart/gas_injection.i

**Reply by MatiasAllay** [reply]
Thanks @cpgr! This solve my question.
